server/service/queries.go

```diff
@@ -479,6 +477,21 @@ func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpe
 				message: fmt.Sprintf("query payload verification: %s", err),
 			})
 		}
+
+		// check that the user can update the query if it already exists
+		query, err := svc.ds.QueryByName(ctx, query.Name)
+		if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return err
+		} else if err == nil {
```

Comment:
I'm not sure if the linter caught this, but if you have a block that ends in a `return`, you can drop the `else` and save some indentation/nesting (i.e. if it goes past the `if`, it is necessarily an `else` because it could not fall through). Not sure how clear my explanation is... :D

Reply:
If I understand correctly, I think you are suggesting this be changed to 

```
		query, err := svc.ds.QueryByName(ctx, query.Name)
		if err != nil && !errors.Is(err, sql.ErrNoRows) {
			return err
		}
		if err == nil {
			if err := svc.authz.Authorize(ctx, query, fleet.ActionWrite); err != nil {
				return err
			}
		}
```

Typically, I like to use either a `switch` or a chain of `else if` s when trying to handle the same error ie anything more complex than `if err != nil`. imo, it make it more clear that you are dealing with the same error.

This is the same reason why I wouldn't rewrite something like

```
switch x {
case Foo:
    return ""
case Bar:
    // something else without return
case Baz:
    // something else without return
}
```

to 

```
if x == Foo {
    return
}

switch x {
    ...
}
```

Happy to discuss further or change it. Luckily the linter is not complaining about this one ;)

Reply:
In this specific case, what I'd do is:

```
if err != nil {
	if errors.Is(sql.ErrNoRows) {
                continue
        }
	return err
}
if err := svc.authz.Authorize(ctx, query, fleet.ActionWrite); err != nil {
	return err
}
// etc...
```

But that's a minor thing, mostly just a preference (and I see that Lucas has a different one too, so YMMV! :D )